packages/utils/crates/utils-db/src/leveldb.rs

```diff
@@ -0,0 +1,206 @@
+use async_trait::async_trait;
+
+use crate::errors::DbError;
+use crate::traits::{AsyncKVStorage, BatchOperation};
+
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+// https://users.rust-lang.org/t/wasm-web-sys-how-to-manipulate-js-objects-from-rust/36504
+#[cfg(feature = "wasm")]
+#[wasm_bindgen]
+extern "C" {
+    pub type LevelDb;
+
+    #[wasm_bindgen(method, catch)]
+    async fn has(this: &LevelDb, key: Box<[u8]>) -> Result<JsValue, JsValue>; // bool;
+
+    #[wasm_bindgen(method, catch)]
+    async fn put(this: &LevelDb, key: Box<[u8]>, value: Box<[u8]>) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(method, catch)]
+    async fn get(this: &LevelDb, key: Box<[u8]>) -> Result<JsValue, JsValue>; // Option<Box<[u8]>>;
+
+    #[wasm_bindgen(method, catch)]
+    async fn remove(this: &LevelDb, key: Box<[u8]>) -> Result<JsValue, JsValue>; // Option<Box<[u8]>>;
+
+    // https://github.com/Level/levelup#dbbatcharray-options-callback-array-form
+    #[wasm_bindgen(method, catch)]
+    async fn batch(this: &LevelDb, operations: js_sys::Array, wait_for_write: bool) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(method, catch)]
+    fn dump(this: &LevelDb, destination: String) -> Result<(), JsValue>;
+}
+
+pub struct LevelDbShim {
+    db: LevelDb,
+}
+
+impl LevelDbShim {
+    pub fn new(db: LevelDb) -> LevelDbShim {
+        LevelDbShim { db }
+    }
+}
+
+#[async_trait(? Send)]
+impl AsyncKVStorage for LevelDbShim {
+    type Key = Box<[u8]>;
+    type Value = Box<[u8]>;
+
+    async fn get(&self, key: Self::Key) -> crate::errors::Result<Self::Value> {
+        self.db
+            .get(key)
+            .await
+            .map_err(|_| DbError::GenericError("Encountered error on DB get operation".to_string()))
+            .and_then(|v| {
+                if v.is_undefined() {
+                    Err(DbError::NotFound)
+                } else {
+                    Ok(js_sys::Uint8Array::new(&v).to_vec().into_boxed_slice())
```

Comment:
Same here: better use `Uint8Array::from(v)`

Reply:
If that works, then you'd simply use `v.into()`

Reply:
it does!